File: app/src/main/java/com/andrespaez/bitrise/ui/factories/GenericAdapterFactory.java
package com.andrespaez.bitrise.ui.factories;

import android.view.ViewGroup;

import com.andrespaez.bitrise.ui.adapters.list.models.GenericItemView;

public abstract class GenericAdapterFactory {

    public static final int TYPE_HEADER = 1001;
    public static final int TYPE_ITEM = 1002;
    public static final int TYPE_FOOTER = 1003;
    public static final int TYPE_CATEGORY = 1004;
    public static final int TYPE_DIVIDER = 1005;

    public GenericAdapterFactory() {
    }

    public abstract GenericItemView onCreateViewHolder(ViewGroup var1, int var2);

}

File: app/src/main/java/com/bitrise/app/ui/factories/GenericAdapterFactory.kt
package com.bitrise.app.ui.factories

import android.view.ViewGroup
import com.bitrise.app.ui.adapters.list.models.GenericItemView

const val ITEM_LIST_TYPE_CATEGORY = 1001

abstract class GenericAdapterFactory {

    abstract fun onCreateViewHolder(var1: ViewGroup, viewType: Int): GenericItemView<*>
}